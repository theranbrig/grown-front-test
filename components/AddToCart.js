import React from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { Button, Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { CURRENT_USER_QUERY } from './User';

const ADD_TO_CART_MUTATION = gql`
  mutation addToCart($id: ID!) {
    addToCart(id: $id) {
      id
      quantity
    }
  }
`;

const AddToCart = ({ id }) => (
  <Mutation
    mutation={ADD_TO_CART_MUTATION}
    variables={{
      id,
    }}
    refetchQueries={[{ query: CURRENT_USER_QUERY }]}
  >
    {(addToCart, { loading }) => (
      <Button icon labelPosition="right" disabled={loading} onClick={addToCart}>
        Add{loading ? 'ing' : ' '} To Cart
        <Icon name="cart plus" />
      </Button>
    )}
  </Mutation>
);

AddToCart.propTypes = {
  id: PropTypes.string.isRequired,
};

export default AddToCart;
export { ADD_TO_CART_MUTATION };
