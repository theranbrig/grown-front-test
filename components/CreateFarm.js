import React, { Component } from 'react';
import { Form, Button, Grid, Icon } from 'semantic-ui-react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Router from 'next/router';
import FormStyling from './styles/FormStyles';
import Error from './ErrorMessage';

// Sign Up User Mutation
const CREATE_FARM_MUTATION = gql`
  mutation CREATE_FARM_MUTATION(
    $name: String!
    $email: String!
    $tagline: String!
    $description: String!
    $location: String!
    $phone: String!
    $website: String!
    $image: String!
  ) {
    createFarm(
      name: $name
      email: $email
      tagline: $tagline
      description: $description
      location: $location
      phone: $phone
      website: $website
      image: $image
    ) {
      id
    }
  }
`;

class CreateFarm extends Component {
  // State for Form
  state = {
    name: '',
    email: '',
    tagline: '',
    description: '',
    location: '',
    phone: '',
    website: '',
    image: '',
  };

  // Enter Information Value Handler
  saveToState = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  uploadFile = async e => {
    const { files } = e.target;
    const data = new FormData();
    data.append('file', files[0]);
    data.append('upload_preset', 'grownphotos');
    const res = await fetch('https://api.cloudinary.com/v1_1/dq7uyauun/image/upload', {
      method: 'POST',
      body: data,
    });
    const file = await res.json();
    console.log(file);
    this.setState({
      image: file.secure_url,
    });
  };

  render() {
    return (
      <Mutation mutation={CREATE_FARM_MUTATION} variables={this.state}>
        {(createFarm, { error, loading }) => {
          if (error) return <Error error={error} />;
          return (
            <Grid>
              <FormStyling>
                <h1>
                  Add Your Farm to <span>GROWN</span>
                </h1>
                <Form
                  success={this.state.completed}
                  method="post"
                  className="create-farm-form"
                  loading={loading}
                  onSubmit={async e => {
                    e.preventDefault();
                    const res = await createFarm();
                    console.log(res);
                    Router.push({
                      pathname: '/farm',
                      query: { id: res.data.createFarm.id },
                    });
                  }}
                >
                  <Form.Group>
                    <Form.Field width={16}>
                      <label htmlFor="name">
                        Farm Name
                        <input
                          minLength={5}
                          maxLength={40}
                          required
                          id="name"
                          type="text"
                          name="name"
                          placeholder="Enter Farm Name"
                          value={this.state.name}
                          onChange={this.saveToState}
                        />
                      </label>
                    </Form.Field>
                  </Form.Group>
                  <Form.Group>
                    <Form.Field width={16}>
                      <label htmlFor="location">
                        Location
                        <input
                          minLength={5}
                          maxLength={60}
                          required
                          id="location"
                          type="text"
                          name="location"
                          placeholder="Enter Farm Address"
                          value={this.state.location}
                          onChange={this.saveToState}
                        />
                      </label>
                    </Form.Field>
                  </Form.Group>
                  <Form.Group>
                    <Form.Field width={16}>
                      <label htmlFor="email">
                        Email Address
                        <input
                          minLength={5}
                          maxLength={100}
                          required
                          id="email"
                          type="email"
                          name="email"
                          placeholder="Enter Farm Email Address"
                          value={this.state.email}
                          onChange={this.saveToState}
                        />
                      </label>
                    </Form.Field>
                  </Form.Group>
                  <Form.Group>
                    <Form.Field width={16}>
                      <label htmlFor="phone">
                        Phone Number
                        <input
                          minLength={9}
                          maxLength={10}
                          required
                          id="phone"
                          type="text"
                          name="phone"
                          placeholder="Add Phone Number"
                          value={this.state.phone}
                          onChange={this.saveToState}
                          max-length="13"
                        />
                      </label>
                    </Form.Field>
                  </Form.Group>
                  <Form.Group>
                    <Form.Field width={16}>
                      <label htmlFor="tagline">
                        Tagline
                        <input
                          minLength={5}
                          maxLength={40}
                          required
                          id="tagline"
                          type="text"
                          name="tagline"
                          placeholder="Add Short Farm Description"
                          value={this.state.tagline}
                          onChange={this.saveToState}
                        />
                      </label>
                    </Form.Field>
                  </Form.Group>
                  <Form.Group>
                    <Form.Field width={16}>
                      <label htmlFor="description">
                        Description - Characters Remaining {500 - this.state.description.length}
                        <textarea
                          minLength={5}
                          maxLength={500}
                          required
                          id="description"
                          rows="4"
                          name="description"
                          placeholder="Write description of Farm - Max 500 Characters"
                          value={this.state.description}
                          onChange={this.saveToState}
                        />
                      </label>
                    </Form.Field>
                  </Form.Group>
                  <Form.Group>
                    <Form.Field width={16}>
                      <label htmlFor="website">
                        Website
                        <input
                          minLength={5}
                          maxLength={40}
                          required
                          id="website"
                          type="text"
                          name="website"
                          placeholder="Enter Farm Website"
                          value={this.state.website}
                          onChange={this.saveToState}
                        />
                      </label>
                    </Form.Field>
                  </Form.Group>
                  <Form.Group>
                    <Form.Field width={16}>
                      <label htmlFor="image">
                        Farm Image
                        <input
                          required
                          id="image"
                          type="file"
                          name="image"
                          placeholder="Choose Farm Image"
                          onChange={this.uploadFile}
                        />
                        {this.state.image && <img width="200" src={this.state.image} alt="Upload Preview" />}
                      </label>
                    </Form.Field>
                  </Form.Group>
                  <Button type="submit" icon labelPosition="right">
                    Add
                    {loading ? 'ing' : ''} Farm
                    <Icon name="right arrow" />
                  </Button>
                </Form>
              </FormStyling>
            </Grid>
          );
        }}
      </Mutation>
    );
  }
}

export default CreateFarm;
