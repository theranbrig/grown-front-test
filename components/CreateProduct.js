import React, { Component } from 'react';
import { Form, Button, Icon, Select, Label, Input } from 'semantic-ui-react';
import styled from 'styled-components';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import PropTypes from 'prop-types';
import Error from './ErrorMessage';
import unitOptions from '../lib/formData';
import { PRODUCTS_QUERY } from './Store';

const CreateProductStyling = styled.div`
  margin: 0 auto;
  button {
    height: 38px;
    background-color: ${props => props.theme.lightBlue} !important;
    margin-top: 23px !important;
  }
  i {
    font-size: 1rem;
  }
`;
const CREATE_PRODUCT_MUTATION = gql`
  mutation CREATE_PRODUCT_MUTATION(
    $name: String!
    $description: String!
    $price: Int!
    $unit: String
    $farmId: String
    $image: String
  ) {
    createProduct(name: $name, description: $description, price: $price, unit: $unit, farmId: $farmId, image: $image) {
      id
    }
  }
`;

class CreateProduct extends Component {
  // State for Form
  state = {
    name: '',
    unit: '',
    price: '',
    description: '',
    farmId: '',
  };

  // Enter Information Value Handler

  componentDidMount() {
    this.setState({ farmId: this.props.id });
  }

  saveToState = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  selectInput = (e, data) => {
    this.setState({ unit: data.value });
  };

  render() {
    return (
      <Mutation
        mutation={CREATE_PRODUCT_MUTATION}
        variables={this.state}
        refetchQueries={[
          {
            query: PRODUCTS_QUERY,
            variables: { farmId: this.state.farmId },
          },
        ]}
      >
        {(createProduct, { error, loading }) => {
          if (error) return <Error error={error} />;
          return (
            <CreateProductStyling>
              <h3>Add Product to Farm</h3>
              <Form
                method="post"
                loading={loading}
                onSubmit={async e => {
                  e.preventDefault();
                  const res = await createProduct();
                  this.setState({
                    name: '',
                    unit: '',
                    price: '',
                    description: '',
                  });
                }}
              >
                <Form.Group>
                  <Form.Field width={4}>
                    <label htmlFor="name">
                      Product Name
                      <input
                        required
                        type="text"
                        name="name"
                        id="name"
                        placeholder="30 Character Limit"
                        maxLength="30"
                        value={this.state.name}
                        onChange={this.saveToState}
                      />
                    </label>
                  </Form.Field>
                  <Form.Field width={6}>
                    <label htmlFor="description">
                      Product Description - {100 - this.state.description.length} Remaining
                      <input
                        required
                        type="text"
                        name="description"
                        id="description"
                        placeholder="100 Character Limit"
                        minLength={5}
                        maxLength={100}
                        value={this.state.description}
                        onChange={this.saveToState}
                      />
                    </label>
                  </Form.Field>
                  <Form.Field width={3}>
                    <label htmlFor="price">
                      Price
                      <Input
                        required
                        labelPosition="right"
                        type="number"
                        name="price"
                        id="price"
                        placeholder="Whole Dollars"
                        maxLength="4"
                      >
                        <Label basic>$</Label>
                        <input value={this.state.price} onChange={this.saveToState} maxLength="4" />
                        <Label>.00</Label>
                      </Input>
                    </label>
                  </Form.Field>
                  <Form.Field
                    id="unit"
                    width={2}
                    control={Select}
                    options={unitOptions}
                    value={this.state.unit}
                    onChange={this.selectInput}
                    label={{ children: 'Units', htmlFor: 'unit' }}
                    search
                    fluid
                    searchInput={{ id: 'unit' }}
                  />
                  <Button type="submit" icon labelPosition="right">
                    Add
                    {loading ? 'ing' : ''} <Icon name="up arrow" />
                  </Button>
                </Form.Group>
              </Form>
            </CreateProductStyling>
          );
        }}
      </Mutation>
    );
  }
}

CreateProduct.propTypes = {
  id: PropTypes.string.isRequired,
};

export default CreateProduct;
