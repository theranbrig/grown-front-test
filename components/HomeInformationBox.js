/* eslint-disable no-undef */
/* eslint-disable global-require */
import React from 'react';
import { Grid, Header, Container, Image } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { InformationBox, ImageBox } from './styles/InformationBoxStyles';

class HomeInformationBox extends React.Component {
  componentDidMount() {
    if (typeof window !== 'undefined') {
      window.WOW = require('wowjs');
    }
    new WOW.WOW({ live: false }).init();
  }

  render() {
    return (
      <Grid.Row centered columns={2}>
        <Grid.Column>
          <Container
            text
            fluid
            textAlign="center"
            className="wow fadeInRight"
            verticalalign="center"
            data-wow-duration="2s"
          >
            <InformationBox>
              <Header as="h2">{this.props.header}</Header>
              <Header as="h3">{this.props.subHeader}</Header>
              <p>{this.props.information}</p>
            </InformationBox>
          </Container>
        </Grid.Column>
        <Grid.Column>
          <ImageBox>
            <Image
              src={this.props.image}
              responsive="true"
              centered
              className="wow fadeInLeft point-image"
              data-wow-duration="2s"
              verticalAlign="middle"
            />
          </ImageBox>
        </Grid.Column>
      </Grid.Row>
    );
  }
}
Grid.propTypes = {
  centered: PropTypes.bool,
  text: PropTypes.bool,
  responsive: PropTypes.bool,
};

HomeInformationBox.propTypes = {
  header: PropTypes.string.isRequired,
  subHeader: PropTypes.string.isRequired,
  information: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default HomeInformationBox;
