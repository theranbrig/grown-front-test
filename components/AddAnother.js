import React from 'react';
import { Mutation } from 'react-apollo';
import { Button, Icon } from 'semantic-ui-react';
import PropTypes from 'prop-types';
import { CURRENT_USER_QUERY } from './User';
import { ADD_TO_CART_MUTATION } from './AddToCart';

const AddAnother = ({ id }) => (
  <Mutation
    mutation={ADD_TO_CART_MUTATION}
    variables={{
      id,
    }}
    refetchQueries={[{ query: CURRENT_USER_QUERY }]}
  >
    {(addToCart, { loading }) => (
      <Button icon disabled={loading} onClick={addToCart}>
        <Icon name="plus" />
      </Button>
    )}
  </Mutation>
);

AddAnother.propTypes = {
  id: PropTypes.string.isRequired,
};

export default AddAnother;
