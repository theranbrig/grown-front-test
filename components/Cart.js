import React from 'react';
import { Query, Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { adopt } from 'react-adopt';
import { Button, Icon, List } from 'semantic-ui-react';
import Link from 'next/link';
import User from './User';
import CartStyles from './styles/CartStyles';
import SignOut from './SignOut';
import RemoveProduct from './RemoveCartProduct';
import AddAnother from './AddAnother';
import RemoveOne from './RemoveOne';
import SubmitPayment from './SubmitPayment';

const LOCAL_STATE_QUERY = gql`
  query {
    cartOpen @client
  }
`;

const TOGGLE_CART_MUTATION = gql`
  mutation {
    toggleCart @client
  }
`;
/* eslint-disable */
const Composed = adopt({
	user: ({ render }) => <User>{render}</User>,
	toggleCart: ({ render }) => <Mutation mutation={TOGGLE_CART_MUTATION}>{render}</Mutation>,
	localState: ({ render }) => <Query query={LOCAL_STATE_QUERY}>{render}</Query>
});
/* eslint-enable */

const Cart = () => (
  <Composed>
    {({ user, toggleCart, localState }) => {
      const { me } = user.data;
      if (!me) return null;
      return (
        <CartStyles open={localState.data.cartOpen}>
          <header>
            <Button icon onClick={toggleCart}>
              <Icon name="delete" />
            </Button>
            <h2>{me.name}'s Cart</h2>
            <p>
              You have <strong>{me.cart.length}</strong> item{me.cart.length === 1 ? '' : 's'} in your cart.
            </p>
          </header>
          <List animated divided>
            {me.cart.map(cartProduct => (
              <List.Item key={cartProduct.id}>
                <List.Header>
                  <Link href={`/farm?id=${cartProduct.product.farm.id}`}>
                    <a>{cartProduct.product.name}</a>
                  </Link>
                </List.Header>
                <List.Description>{cartProduct.product.farm.name}</List.Description>
                <p>
                  ${cartProduct.product.price} &times; {cartProduct.quantity} {cartProduct.product.unit}
                  {cartProduct.quantity === 1 ? '' : 's'} ={' '}
                  <strong>
                    ${cartProduct.product.price * cartProduct.quantity}
                    .00
                  </strong>
                </p>
                <div className="cart-item-buttons">
                  <RemoveOne id={cartProduct.product.id} />
                  <AddAnother id={cartProduct.product.id} />
                  <RemoveProduct id={cartProduct.id} />
                </div>
              </List.Item>
            ))}
          </List>
          <footer>
            <h3>
              Total: $
              {me.cart.reduce((tally, cartProduct) => {
                if (!cartProduct.product) return tally;
                return tally + cartProduct.quantity * cartProduct.product.price;
              }, 0)}
            </h3>
          </footer>
          <SubmitPayment>
            <Button icon labelPosition="right" className="buy-button">
              Buy Now
              <Icon name="cart plus" />
            </Button>
          </SubmitPayment>
          <SignOut />
        </CartStyles>
      );
    }}
  </Composed>
);

export default Cart;
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION };
