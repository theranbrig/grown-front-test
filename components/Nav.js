import React from 'react';
import Link from 'next/link';
import { Mutation } from 'react-apollo';
import { Icon } from 'semantic-ui-react';
import NavStyles from './styles/NavStyles';
import User from './User';
import { TOGGLE_CART_MUTATION } from './Cart';

const Nav = () => (
  <User>
    {({ data: { me } }) => (
      <NavStyles className="navigation-bar">
        <Link href="/">
          <a>Home</a>
        </Link>
        <Link href="/browse">
          <a>Farms</a>
        </Link>
        {me && (
          <React.Fragment>
            <Link href="/orders">
              <a>Orders</a>
            </Link>
            <Mutation mutation={TOGGLE_CART_MUTATION}>
              {toggleCart => (
                <button onClick={toggleCart} type="button">
                  <Icon name="shopping cart" /> {me.cart.reduce((tally, product) => tally + product.quantity, 0)}
                </button>
              )}
            </Mutation>
          </React.Fragment>
        )}
        {/* <Link href="/markets">
			<a>Markets</a>
			</Link>
			<Link href="/cart">
			<a>Cart</a>
		</Link> */}
        {!me && (
          <Link href="/login">
            <a>Login</a>
          </Link>
        )}
      </NavStyles>
    )}
  </User>
);

export default Nav;
