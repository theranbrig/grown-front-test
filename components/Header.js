import Router from 'next/router';
import NProgress from 'nprogress';
import Nav from './Nav';
import Cart from './Cart';
import User from './User';
import StyledHeader, { Logo } from './styles/HeaderStyles';

Router.onRouteChangeStart = () => {
  NProgress.start();
};

Router.onRouteChangeComplete = () => {
  NProgress.done();
};

Router.onRouteChangeError = () => {
  NProgress.done();
};

const Header = () => (
  <StyledHeader>
    <div>
      <Logo className="bar">
        <div>
          <a href="/">
            <img width="150" src="https://i.imgur.com/Gcwggjm.png" alt="Grown Logo" />
          </a>
        </div>
        <Nav />
      </Logo>
    </div>
    <User>{({ data: { me } }) => me && <Cart />}</User>
  </StyledHeader>
);

export default Header;
