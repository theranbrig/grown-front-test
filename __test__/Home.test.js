import { shallow, mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import Header from '../components/Header';
import HomeInformationBox from '../components/HomeInformationBox';
import Nav from '../components/Nav';
import { CURRENT_USER_QUERY } from '../components/User';

const fakeBox = {
  header: 'Title',
  subHeader: 'More Words',
  information: 'Writing',
  image: 'picture.jpg',
};

const notSignedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: null } },
  },
];

describe('Header', () => {
  it('should render the Header Logo', () => {
    const wrapper = shallow(<Header />);
    const img = wrapper.find('img');
    expect(img.props().alt).toContain('Grown Logo');
  });
});

describe('NavBar', () => {
  it('should render the Navbar', async () => {
    const wrapper = mount(
      <MockedProvider mocks={notSignedInMocks}>
        <Nav />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const nav = wrapper.find('.navigation-bar');
    expect(toJSON(nav)).toMatchSnapshot();
  });
});

describe('Home Information Box', () => {
  it('should render the Home Information Box', async () => {
    const wrapper = mount(
      <MockedProvider mocks={notSignedInMocks}>
        <HomeInformationBox header={fakeBox.header} image={fakeBox.image} />
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    const Header = wrapper.find('Header');
    expect(toJSON(Header)).toMatchSnapshot();
    const img = wrapper.find('Image');
    expect(toJSON(img)).toMatchSnapshot();
  });
});
