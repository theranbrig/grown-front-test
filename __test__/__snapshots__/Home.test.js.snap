// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Information Box should render the Home Information Box 1`] = `
Array [
  <Header
    as="h2"
  >
    <h2
      className="ui header"
    >
      Title
    </h2>
  </Header>,
  <Header
    as="h3"
  >
    <h3
      className="ui header"
    />
  </Header>,
]
`;

exports[`Home Information Box should render the Home Information Box 2`] = `
<Image
  as="img"
  centered={true}
  className="wow fadeInLeft point-image"
  data-wow-duration="2s"
  responsive="true"
  src="picture.jpg"
  ui={true}
  verticalAlign="middle"
>
  <img
    className="ui centered middle aligned image wow fadeInLeft point-image"
    data-wow-duration="2s"
    responsive="true"
    src="picture.jpg"
  />
</Image>
`;

exports[`NavBar should render the Navbar 1`] = `
Array [
  <NavStyles
    className="navigation-bar"
  >
    <ul
      className="navigation-bar NavStyles-sc-1uctv9x-0 kAIeCk"
    >
      <Link
        href="/"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </Link>
      <Link
        href="/browse"
      >
        <a
          href="/browse"
          onClick={[Function]}
        >
          Farms
        </a>
      </Link>
      <Link
        href="/login"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </Link>
    </ul>
  </NavStyles>,
  <ul
    className="navigation-bar NavStyles-sc-1uctv9x-0 kAIeCk"
  >
    <Link
      href="/"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </Link>
    <Link
      href="/browse"
    >
      <a
        href="/browse"
        onClick={[Function]}
      >
        Farms
      </a>
    </Link>
    <Link
      href="/login"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </Link>
  </ul>,
]
`;
